cli/client.go

```diff
@@ -380,6 +387,83 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// CheckUpdateResponse holds the values used to hold release information.
+type CheckUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdate is the corresponding Action for 'check-update'.
+func CheckUpdate(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	infof(c.App.ErrWriter, "checking if update available")
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		infof(c.App.ErrWriter, "Failed to pare dateCompiled")
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("failed to parse compiled date: %w", err)
```
The CLI will automatically print returned errors IIRC, so no need to log it and return it.

Reply:
sounds good

Reply:
The benefit of logging it and returning nil is that it doesn't block the cli from working.

If anything, I suspect we should fail silently here

Reply:
I would agree that we should fail silently (should only log if we're in some sort of verbose mode which I don't think we have) - ultimately this check is not absolutely needed for operation

Reply:
Ah yeah SGTM on failing silently.